pkg/orchestration/wiring/datadog/plugin.go

```diff
@@ -0,0 +1,201 @@
+	"github.com/atlassian/voyager/pkg/orchestration/wiring/wiringutil/oap"
+	"github.com/atlassian/voyager/pkg/orchestration/wiring/wiringutil/svccatentangler"
+	sc_v1b1 "github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/v1beta1"
+	"github.com/pkg/errors"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const (
+	clusterServiceClassExternalID                      = "875d4a87-e887-4838-a0b5-b64491dbf9cb"
+	clusterServicePlanExternalID                       = "d8048a2d-49de-4fda-b7ef-328de171cd32"
+	ResourceType                  voyager.ResourceType = "datadog-alarm"
+)
+
+type WiringPlugin struct {
+	svccatentangler.SvcCatEntangler
+}
+
+var (
+	serviceInstanceGVK = sc_v1b1.SchemeGroupVersion.WithKind("ServiceInstance")
+)
+
+func WireUp(stateResource *orch_v1.StateResource, context *wiringplugin.WiringContext) (*wiringplugin.WiringResult, bool, error) {
+	err := validateRequest(stateResource, context)
+	if err != nil {
+		return nil, false, err
+	}
+	kubeComputeDependency, err := context.TheOnlyDependency()
+	if err != nil {
+		return nil, false, err
+	}
+	setOfScalingShape, found, err := knownshapes.FindSetOfPScalingShape(kubeComputeDependency.Contract.Shapes)
+	if err != nil {
+		return nil, false, err
+	}
+	if !found {
+		return nil, false, errors.Errorf("failed to find shape %q in contract of %q", knownshapes.SetOfScalingShape, kubeComputeDependency.Name)
+	}
+
+	var wiredResources []smith_v1.Resource
+
+	deploymentResourceName := setOfScalingShape.Data.DeploymentResourceName
+	CPUServiceInstance, err := constructServiceInstance(stateResource, context, deploymentResourceName, CPU)
+	if err != nil {
+		return nil, false, err
+	}
+	wiredResources = append(wiredResources, CPUServiceInstance)
+
+	memoryServiceInstance, err := constructServiceInstance(stateResource, context, deploymentResourceName, Memory)
+	if err != nil {
+		return nil, false, err
+	}
+	wiredResources = append(wiredResources, memoryServiceInstance)
+
+	result := &wiringplugin.WiringResult{
+		Resources: wiredResources,
+	}
+
+	return result, false, nil
+}
+
+func constructServiceInstance(resource *orch_v1.StateResource, context *wiringplugin.WiringContext, deploymentResourceName smith_v1.ResourceName, alarmType AlarmType) (smith_v1.Resource, error) {
+	instanceID, err := svccatentangler.InstanceID(resource.Spec)
+	if err != nil {
+		return smith_v1.Resource{}, err
+	}
+	threshold := AlarmThresholds{
+		Critical: 90,
+		Warning:  80,
+	}
+	query := generateQuerySpec(context, alarmType, threshold)
+	alarmsAtt := createFinalAlarmSpec(resource, &context.StateContext, query)
+	if err != nil {
+		return smith_v1.Resource{}, err
+	}
+	resourceName, err := oap.ResourceName(resource.Spec)
+	if err != nil {
+		return smith_v1.Resource{}, err
+	}
+	if resourceName == "" {
+		resourceName = string(resource.Name)
+	}
+
+	resourceName = resourceName + "--" + string(alarmType)
+	serviceInstanceSpec := ServiceInstanceSpec{
```

Comment:
Is this really enough to create an alarm? `Label` and resource name of the compute are not taken into account here. i.e. will it work if we have multiple compute resources in SD?